src/SBarcodeScanner.cpp

```diff
@@ -23,14 +24,20 @@ SBarcodeScanner::~SBarcodeScanner()
     stopCam();
 }
 
-SBarcodeDecoder *SBarcodeScanner::getDecoder()
+SBarcodeDecoder* SBarcodeScanner::getDecoder()
 {
     return &m_decoder;
 }
 
-void SBarcodeScanner::initCam() {
+void SBarcodeScanner::initCam()
+{
     camera = new QCamera(this);
 
+    if (camera->error()) {
+        qDebug() << "Error during camera initialization: " << camera->errorString();
```

Comment:
Also maybe a good idea would be to add `error` property, with error description to use in QML